### ISSUE TITLE
`ask_username` option for ssh connections

### ISSUE THREAD

**Issue by darshan-hindocha** [iterative/dvc#8942]
Currently if you want to hook up dvc to a fileserver you have to provide something like the following:

```
[core]
    remote = ssh-storage
['remote "ssh-storage"']
    url = ssh://path.to/model_registry
    port = 22
    user = user123
    ask_password = true
```

This means that I only have to put my password in when I run commands that require dvc to access the fileserver.

However, since this config file is checked in to git and used by others I would find it useful to have an `ask_username` option like the `ask_password` flag. This means everyone can share this config file without having to change it. Currently you have to change the username, creating a diff on the config file that you ignore. (you can't gitignore this file as it's needed in the project)

TLDR; enable an `ask_user` field for ssh connections
